planner/core/planbuilder.go

```diff
@@ -1588,6 +1590,54 @@ func (b *PlanBuilder) buildLoadStats(ld *ast.LoadStatsStmt) Plan {
 	return p
 }
 
+func (b *PlanBuilder) buildSplitIndexRegion(node *ast.SplitIndexRegionStmt) (Plan, error) {
+	tblInfo := node.Table.TableInfo
+	indexInfo := tblInfo.FindIndexByName(strings.ToLower(node.IndexName))
+	if indexInfo == nil {
+		return nil, ErrKeyDoesNotExist.GenWithStackByArgs(node.IndexName, tblInfo.Name)
+	}
+	schema := expression.TableInfo2SchemaWithDBName(b.ctx, node.Table.Schema, tblInfo)
+	mockTablePlan := LogicalTableDual{}.Init(b.ctx)
+	mockTablePlan.SetSchema(schema)
+
+	tps := make([]*types.FieldType, 0, len(indexInfo.Columns))
+	for _, c := range indexInfo.Columns {
+		col := tblInfo.Columns[c.Offset]
+		tps = append(tps, &col.FieldType)
+	}
+
+	indexValues := make([][]types.Datum, 0, len(node.ValueLists))
+	for i, valuesItem := range node.ValueLists {
+		if len(valuesItem) > len(indexInfo.Columns) {
+			return nil, ErrWrongValueCountOnRow.GenWithStackByArgs(i + 1)
+		}
+		valueList := make([]types.Datum, 0, len(valuesItem))
+		for j, valueItem := range valuesItem {
+			x, ok := valueItem.(*driver.ValueExpr)
+			if !ok {
+				return nil, errors.Trace(errors.New("expect constant values"))
+			}
+			value, err := x.Datum.ConvertTo(b.ctx.GetSessionVars().StmtCtx, tps[j])
```

Comment:
tps[j] can be `tblInfo.Columns[j].FieldType`

Reply:
`tblInfo.Columns[indexInfo.Columns[j].Offset]`
done.